### PR TITLE
Add Group TEs setting into the settings screen

### LINKED ISSUE #4217: Add a user settings for grouped Time Entries
🏆 User should be able to decide whether they want their time entries to be grouped or not from the Settings screen. Make sure that we save the value received from `me/preferences` endpoint and use it in the viewmodel.

### PR DESCRIPTION
**Pull request by kornelijepetak**
## What's this?
Adds `group similar time entries` to Settings screen.

### Relationships
Closes #4217 

## Why do we want this?
Users want customization. Also, 🎡wheels.

## How is it done?
The usuals.

### Why not another way?
Read previous point. Consistency.

### Side effects
None noticed.
I have zero ideas why xcode messed up the `.designer.cs` file. I believe it's fine.

## Review hints
Commit by commit makes most sense:
* Foundation
* iOS
* Android
* Test

## :squid: Permissions
Me and this guy @simonrozsival.

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/Settings/SettingsViewModel.cs [hunk 1]
```diff
@@ -191,6 +193,11 @@ public sealed class SettingsViewModel : MvxViewModel
                     .Select(preferences => preferences.TimeOfDayFormat.IsTwentyFourHoursFormat)
                     .DistinctUntilChanged();
 
+            IsGroupingTimeEntries =
+                dataSource.Preferences.Current
+                    .Select(preferences => preferences.CollapseTimeEntries)
+                    .DistinctUntilChanged();
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/Settings/SettingsViewModel.cs]: This should be a driver. I think that might explain why you were having some problems with syncing. All the others should be drivers as well.

File: Toggl.Daneel/ViewControllers/Settings/SettingsViewController.cs [hunk 2]
```diff
@@ -130,6 +131,10 @@ public override void ViewDidLoad()
                 .Subscribe(ViewModel.ToggleManualMode)
                 .DisposedBy(DisposeBag);
 
+            GroupSimilarTimeEntriesSwitch.Rx().Changed()
+                .Subscribe(ViewModel.ToggleTimeEntriesGrouping.Inputs)
```
Comment (simonrozsival) [inline Toggl.Daneel/ViewControllers/Settings/SettingsViewController.cs]: `BindAction` can't be used here?

File: Toggl.Foundation.MvvmCross/ViewModels/Settings/SettingsViewModel.cs (hunk 1 above)
Comment (kornelijepetak) [reply Toggl.Foundation.MvvmCross/ViewModels/Settings/SettingsViewModel.cs]: I turned this into a driver. I would also rather not change the rest in this PR as it is unrelated. Also, coming with RxFactoring I think.

File: Toggl.Daneel/ViewControllers/Settings/SettingsViewController.cs (hunk 2 above)
Comment (kornelijepetak) [reply Toggl.Daneel/ViewControllers/Settings/SettingsViewController.cs]: I introduced a `BindAction` on `UISwitch` so yes, it can 😄

**Pull request merged**